library/src/main/java/com/bumptech/glide/load/resource/bitmap/BitmapDrawLock.java

```diff
@@ -0,0 +1,81 @@
+package com.bumptech.glide.load.resource.bitmap;
+
+import android.os.Build;
+import android.support.annotation.NonNull;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * https://github.com/bumptech/glide/issues/738 On some devices (Moto X with android 5.1) bitmap
+ * drawing is not thread safe.
+ *
+ * This class provides a lock implementation that only really locks for these devices. For other
+ * types of devices the lock is always available and therefore does not impact performance The
+ * actual lock logic is delegated to {@see ReentrantLock}
+ */
+final class BitmapDrawLock implements Lock {
```

Comment:
How about?
```java
final class BitmapDrawLock {
	// only enable the lock for Moto X 2nd gen on android 5.1
	private static final Lock INSTANCE =
			"XT1097".equals(Build.MODEL) && Build.VERSION.SDK_INT == Build.VERSION_CODES.LOLLIPOP_MR1
			? new ReentrantLock()
			: new NoLock();

	private BitmapDrawLock() {
		// static utility class
	}

	public static Lock getInstance() {
		return INSTANCE;
	}
}

public class NoLock implements Lock {
	@Override public void lock() {
		// do nothing
	}
	@Override public void lockInterruptibly() throws InterruptedException {
		// do nothing
	}
	@Override public boolean tryLock() {
		return true;
	}
	@Override public boolean tryLock(long time, @NotNull TimeUnit unit) throws InterruptedException {
		return true;
	}
	@Override public void unlock() {
		// do nothing
	}
	@NotNull @Override public Condition newCondition() {
		throw new UnsupportedOperationException("Should not be called");
	}
}
```
That single INSTANCE field + getInstance can be even moved to some other class, like `TransformationUtils`.